ArrCharOps.java

```diff
@@ -126,7 +179,61 @@ public static long hashCode(char[] arr) {
      *         return -2 if there is an error with the input.
      */
     public static int compareTo(String str1, String str2) {
-        // Replace the following statement with your code
-        return 0;
-    }
+        if (str1.isEmpty()||str2.isEmpty()) {
+            return -2;
+        }
+        int counterequal = 0 ;
+        if (str1.length()>str2.length()) {
+           for(int i = 0 ; i< str2.length(); i++) 
+           {
+            if (str1.charAt(i)==str2.charAt(i)) {
+                counterequal++;
+            }
+            if (str1.charAt(i)>str2.charAt(i)){
+                return 1;
+            }
+            if (str1.charAt(i)<str2.charAt(i)) {
+                return -1;
+            }
+           }
+           if (counterequal == str2.length()) {
+                {
+                   return 1;
+                }
+           }
+        }
+        if (str1.length()<str2.length()) {
+            for(int i = 0 ; i< str1.length(); i++) 
+            {
+             if (str1.charAt(i)==str2.charAt(i)) {
+                 counterequal++;
+             }
+             if (str1.charAt(i)>str2.charAt(i)){
+                 return 1;
+             }
+             if (str1.charAt(i)<str2.charAt(i)) {
+                 return -1;
+             }
+            }
+                    return -1;
+            }
+            // now it means the length is equal
+            for(int i = 0 ; i< str2.length(); i++) 
+            {
+             if (str1.charAt(i)==str2.charAt(i)) {
+                 counterequal++;
+             }
+             if (str1.charAt(i)>str2.charAt(i)){
+                 return 1;
+             }
+             if (str1.charAt(i)<str2.charAt(i)) {
+                 return -1;
+             }
+            }
+                 return 0;
+        
```

Comment:
1. Repeated code.
2 .  try and structure your code prior to starting the solution. 
```
int minLength = Math.min(str1.length(), str2.length());
for (int i = 0; i < minLength; i++) {
            char ch1 = str1.charAt(i);
            char ch2 = str2.charAt(i);
            if (ch1 != ch2){
                 return ch1 > ch2 ? 1 : -1;
            }
}
if (str1.length() != str2.length()){
    return str1.length() > str2.length() ? 1 : -1;
}
return 0; 
```